risc0/zkvm/src/host/api/server.rs

```diff
@@ -835,12 +833,27 @@ fn execute_redis(
     exec: &mut ExecutorImpl,
     params: super::RedisParams,
 ) -> Result<crate::Session> {
-    let client = redis::Client::open(params.url)?;
-    let mut connection = client.get_connection()?;
-    exec.run_with_callback(|segment| {
+    let (sender, receiver) = kanal::unbounded();
```

Comment:
Might want to bound this, unclear what the bound should be however.

Reply:
I'll test a few values, but the [crate's benchmark](https://github.com/fereidani/rust-channel-benchmarks) didn't show a big improvement vs unbounded.

Reply:
[d90aaba](https://github.com/risc0/risc0/pull/2407/commits/d90aaba376ae3960dbc49a422102e0abca8dca56)

I set the size to 100, to mimic what we have in Bonsai. I didn't notice a difference in performance between a bound of 10, 50, and 100.

Reply:
I would suggest have the default be override-able via env vars so we can tune it externally to our machine memory IMHO.

Reply:
[Done](https://github.com/risc0/risc0/pull/2407/commits/6ba76b94c7ea0cffebb7ed824290c7fcf4ffd962#diff-6a4cb9d6d6a46162414dc817ee4e765428b256aa3307d039484053b40cbb800cR836-R840)